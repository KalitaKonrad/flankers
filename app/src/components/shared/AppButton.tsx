import React from 'react';
import { StyleSheet } from 'react-native';
import { Button } from 'react-native-paper';

type AppButtonProps = React.ComponentProps<typeof Button>;

export const AppButton: React.FC<AppButtonProps> = (props) => (
  <Button
    {...props}
    theme={{ roundness: 100 }}
    uppercase={false}
    contentStyle={[
      !props.compact && styles.buttonContainer,
      props.contentStyle,
    ]}
    labelStyle={[
      styles.text,
      props.mode === 'contained' && styles.textInverted,
      props.labelStyle,
    ]}
  />
);

const styles = StyleSheet.create({
  buttonContainer: {
    height: 56,
  },
  text: {
    letterSpacing: 0,
    fontWeight: 'bold',
    fontSize: 16,
  },
  textInverted: {
    color: '#fff',
  },
});
