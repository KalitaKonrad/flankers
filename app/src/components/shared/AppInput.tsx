import React, { ComponentProps } from 'react';
import { TextInput } from 'react-native-paper';

export type AppInputProps = ComponentProps<typeof TextInput>;

export const AppInput: React.FC<AppInputProps> = (props) => (
  <TextInput {...props} theme={{ roundness: 8 }} />
);
