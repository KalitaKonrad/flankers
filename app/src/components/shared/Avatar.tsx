import * as React from 'react';
import { Image, ImageSourcePropType, StyleSheet, View } from 'react-native';

import { theme } from '../../theme';

interface AvatarProps {
  size: number;
  src: ImageSourcePropType;
  border?: number;
  elevation?: number;
  borderRadius?: number;
}

export const Avatar: React.FC<AvatarProps> = ({
  size,
  src,
  border = 0,
  elevation = 0,
  borderRadius = 100,
}) => (
  <View
    style={[
      styles.container,
      {
        elevation,
        height: size,
        width: size,
        borderRadius,
      },
    ]}>
    <Image
      style={[
        styles.avatar,
        {
          height: size,
          width: size,
          borderWidth: border,
          borderRadius,
        },
      ]}
      source={src}
    />
  </View>
);

const styles = StyleSheet.create({
  container: {
    zIndex: 80,
    backgroundColor: '#fff',
  },
  avatar: {
    borderColor: theme.colors.white,
  },
});
