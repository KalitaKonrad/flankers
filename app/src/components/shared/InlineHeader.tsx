import React from 'react';
import { StyleSheet, View } from 'react-native';

interface InlineHeaderProps {
  color?: string;
  center?: boolean;
}

export const InlineHeader: React.FC<InlineHeaderProps> = ({
  children,
  color,
  center,
}) => {
  return (
    <View
      style={[
        styles.container,
        {
          backgroundColor: color,
          justifyContent: center ? 'center' : 'space-between',
        },
      ]}>
      {children}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'row',
    position: 'relative',
    padding: 10,
  },
});
