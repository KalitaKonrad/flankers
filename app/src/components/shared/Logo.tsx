import * as React from 'react';
import { Image } from 'react-native';

export const Logo: React.FC<
  Omit<React.ComponentProps<typeof Image>, 'source'>
> = (props) => (
  <Image {...props} source={require('../../../assets/flankers-logo.png')} />
);
