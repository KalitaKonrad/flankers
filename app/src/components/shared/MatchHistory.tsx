import React from 'react';
import { List } from 'react-native-paper';

import { MatchResponse } from '../../types/matchResponse';

interface MatchHistoryProps {
  name: string;
  matchHistory: MatchResponse[];
}

export const MatchHistory: React.FC<MatchHistoryProps> = ({
  name,
  matchHistory,
}) => {
  return (
    <List.Item
      title="Match"
      description="Result"
      left={(props) => <List.Icon {...props} icon="star" />}
    />
  );
};
