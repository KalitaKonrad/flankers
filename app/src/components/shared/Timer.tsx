import React, { useEffect, useState } from 'react';

import RoundInformation from './RoundInformation';

interface TimerProps {
  subText?: string;
}

const formatSeconds = (seconds: number) => {
  const minutes = ('0' + Math.floor(seconds / 60)).slice(-2);
  const secondsMod = ('0' + (seconds % 60)).slice(-2);

  return `${minutes}:${secondsMod}`;
};

export const Timer: React.FC<TimerProps> = ({ subText }) => {
  const [seconds, setSeconds] = useState(0);

  useEffect(() => {
    const interval = setInterval(() => {
      setSeconds((seconds) => seconds + 1);
    }, 1000);
    return () => clearInterval(interval);
  }, []);

  return (
    <RoundInformation mainText={formatSeconds(seconds)} subText={subText} />
  );
};
