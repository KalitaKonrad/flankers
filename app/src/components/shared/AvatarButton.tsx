import * as ImagePicker from 'expo-image-picker';
import React, { useEffect, useState } from 'react';
import { Platform, Text } from 'react-native';
import { useTheme } from 'react-native-paper';

import { useUpdateAvatarMutation } from '../../hooks/useUpdateAvatarMutation';
import { useUpdateTeamAvatarMutation } from '../../hooks/useUpdateTeamAvatarMutation';
import { AppButton } from './AppButton';

interface AvatarButtonProps {
  avatarUri: string;
  onAvatarChange(uri: string): void;
  isTeamAvatar: boolean;
  teamId?: string;
}

export const AvatarButton: React.FC<AvatarButtonProps> = (props) => {
  const theme = useTheme();

  const [avatar, setAvatar] = useState<string>(props.avatarUri);
  const [mutateAvatar, mutationAvatar] = useUpdateAvatarMutation();
  const [mutateTeamAvatar, mutationTeamAvatar] = useUpdateTeamAvatarMutation();

  useEffect(() => {
    (async () => {
      if (Platform.OS !== 'web') {
        const {
          status,
        } = await ImagePicker.requestCameraRollPermissionsAsync();
        if (status !== 'granted') {
          alert('Sorry, we need camera roll permissions to make this work!');
        }
      }
    })();
  });

  const onAvatarButtonClick = async () => {
    const result = await ImagePicker.launchImageLibraryAsync({
      mediaTypes: ImagePicker.MediaTypeOptions.All,
      allowsEditing: true,
      aspect: [4, 3],
      quality: 1,
    });
    console.log(result);
    if (!result.cancelled) {
      setAvatar(result.uri);
      if (!props.isTeamAvatar) {
        props.onAvatarChange(result.uri);
        mutateAvatar(result.uri);
        return;
      }
      if (props.teamId !== undefined) {
        props.onAvatarChange(result.uri);
        mutateTeamAvatar({ avatarUri: result.uri, team_id: props.teamId });
      }
    }
  };

  return (
    <AppButton
      mode="contained"
      style={{
        backgroundColor: theme.colors.secondary,
        width: '30%',
        position: 'absolute',
        zIndex: 500,
      }}
      onPress={onAvatarButtonClick}>
      Zmie≈Ñ
    </AppButton>
  );
};
