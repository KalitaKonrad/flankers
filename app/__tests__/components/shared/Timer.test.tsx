import { render } from '@testing-library/react-native';
import * as React from 'react';

import { Timer } from '../../../src/components/shared/Timer';

describe('<Timer>', () => {
  it('renders correctly', () => {
    const tree = render(<Timer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
