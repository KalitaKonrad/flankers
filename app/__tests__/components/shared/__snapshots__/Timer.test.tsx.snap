// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timer> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#ffaf19",
      "borderRadius": 100,
      "borderWidth": 3,
      "height": 196,
      "justifyContent": "center",
      "position": "relative",
      "width": 196,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 30,
          "fontWeight": "bold",
        },
        Object {
          "color": "#ffaf19",
          "fontWeight": "bold",
        },
      ]
    }
    variant="h1"
  >
    00:00
  </Text>
</View>
`;
